File: modern/_1.15.2/src/main/java/cn/enaium/noexpensive/callback/impl/AnvilSetOutputCallbackImpl.java
package cn.enaium.noexpensive.callback.impl;

import cn.enaium.noexpensive.callback.AnvilSetOutputCallback;
import net.minecraft.item.ItemStack;
import net.minecraft.nbt.CompoundTag;
import net.minecraft.nbt.ListTag;
import net.minecraft.nbt.StringTag;
import net.minecraft.text.Text;
import net.minecraft.text.TranslatableText;
import net.minecraft.util.Formatting;

/**
 * @author Enaium
 */
public class AnvilSetOutputCallbackImpl implements AnvilSetOutputCallback {
    @Override
    public void interact(ItemStack output, int levelCost, boolean canTake) {
        if (!output.equals(ItemStack.EMPTY)) {
            if (output.getTag() != null) {
                CompoundTag compoundTag = output.getOrCreateSubTag("display");
                compoundTag.put("Lore", new ListTag());
                compoundTag.getList("Lore", 9).add(StringTag.of(Text.Serializer.toJson(new TranslatableText("container.repair.cost", levelCost).styled(style -> style.setBold(true).setColor(canTake ? Formatting.GREEN : Formatting.RED)))));
            }
        }
    }
}

File: modern/_1.15.2/src/main/kotlin/cn/enaium/noexpensive/callback/impl/AnvilSetOutputCallbackImpl.kt
package cn.enaium.noexpensive.callback.impl

import cn.enaium.noexpensive.callback.AnvilSetOutputCallback
import net.minecraft.client.MinecraftClient
import net.minecraft.entity.player.PlayerEntity
import net.minecraft.item.ItemStack
import net.minecraft.nbt.ListTag
import net.minecraft.nbt.StringTag
import net.minecraft.text.Style
import net.minecraft.text.Text
import net.minecraft.text.TranslatableText
import net.minecraft.util.Formatting

/**
 * @author Enaium
 */
class AnvilSetOutputCallbackImpl : AnvilSetOutputCallback {
    override fun interact(output: ItemStack, levelCost: Int, canTake: Boolean, player: PlayerEntity) {
        MinecraftClient.getInstance().player == player || player.abilities.creativeMode && return
        if (output != ItemStack.EMPTY) {
            if (output.tag != null) {
                val compoundTag = output.getOrCreateSubTag("display")
                compoundTag.put("Lore", ListTag())
                compoundTag.getList("Lore", 9).add(
                    StringTag.of(
                        Text.Serializer.toJson(
                            TranslatableText("container.repair.cost", levelCost).styled { style: Style ->
                                style.setBold(true).setColor(if (canTake) Formatting.GREEN else Formatting.RED)
                            })
                    )
                )
            }
        }
    }
}
